File: Top_Interview_150/Array_String/cpp/80.cpp
#include <vector>

using namespace std;

class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        int index = 1;
        bool flag = false;
        for (int i = 1; i < nums.size(); ++i)
        {
            if (nums[i] != nums[i - 1])
            {
                flag = false;
                nums[index] = nums[i];
                index++;
            }
            else
            {
                if (!flag)
                {
                    flag = true;
                    nums[index] = nums[i];
                    index++;
                }
            }
        }

        return index;
    }
};

File: Top_Interview_150/Array_String/python/80.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        index = 1
        flag = False
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                flag = False
                nums[index] = nums[i]
                index += 1
            else:
                if not flag:
                    flag = True
                    nums[index] = nums[i]
                    index += 1

        return index
